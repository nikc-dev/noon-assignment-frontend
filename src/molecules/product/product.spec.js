import React from 'react';
import { shallow,  configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from 'enzyme-to-json';
import Product from './index';

configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe ( 'Should test the product component', () =>{
  it('renders correctly', () => {
    const component = shallow(
      <Product />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should trigger the update function on click', () => {
    let updateLiked = jest.fn();
    const component = shallow(
      <Product updateLiked={updateLiked} />
    );
    component.find('#product-btn').simulate('click')
    expect(updateLiked).toHaveBeenCalled();
  });

  it('should trigger the delete function on click', () => {
    let updateLiked = jest.fn();
    const component = shallow(
      <Product updateLiked={updateLiked} liked={true} parentIsLiked={true} />
    );
    component.find('#product-btn').simulate('click')
    expect(updateLiked).toHaveBeenCalled();
  });

  it('should disable the button when liked and parent is home', () => {
    let updateLiked = jest.fn();
    const component = shallow(
      <Product updateLiked={updateLiked} liked={true} parentIsLiked={false} />
    );
    component.find('#product-btn').simulate('click')
    expect(updateLiked).toHaveBeenCalled();
  });
})
