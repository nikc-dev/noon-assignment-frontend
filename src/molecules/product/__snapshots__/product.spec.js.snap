// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should test the product component renders correctly 1`] = `
<div
  className="product"
>
  <img
    className="product__img"
  />
  <p
    className="product__name"
  />
  <p
    className="product__description"
  />
  <Container
    className="product__like-container"
    fluid={false}
  >
    <Row
      noGutters={false}
    >
      <Col
        className="product__left-col"
      >
        <img
          alt="express"
          className="product__express-img"
          src="https://z.nooncdn.com/s/app/com/noon/images/fulfilment_express_v2-en.svg"
        />
      </Col>
      <Col
        className="product__right-col"
      >
        <button
          className="product__like-btn"
          id="product-btn"
          onClick={[Function]}
          type="button"
        >
          Like
        </button>
      </Col>
    </Row>
  </Container>
</div>
`;
